src/material/datepicker/datepicker.ts

```diff
@@ -210,8 +210,21 @@ export class MatDatepicker<D> implements OnDestroy, CanColor {
    */
   @Output() readonly monthSelected: EventEmitter<D> = new EventEmitter<D>();
 
-  /** Classes to be passed to the date picker panel. Supports the same syntax as `ngClass`. */
-  @Input() panelClass: string | string[];
+  private _panelClass: string[];
+
+  /** Classes to be passed to the date picker panel. */
+  get panelClass(): string | string[] {
+      return this._panelClass;
+  }
+
+  /**
+   * Classes to be passed to the date picker panel.
+   * Supports string and string array values, similar to `ngClass`.
+   */
+  @Input()
+  set panelClass(value: string | string[]) {
+      this._panelClass = coerceStringArray(value);
+  }
```

Comment:
We generally put `@Input` on the getter and arrange the code like this:
```ts
  /**
   * Classes to be passed to the date picker panel.
   * Accepts string and string array values, similar to `ngClass`.
   */
  @Input()
  get panelClass(): string | string[] { return this._panelClass; }
  set panelClass(value: string | string[]) {
    this._panelClass = coerceStringArray(value);
  }
  private _panelClass: string[];
```

Reply:
Thanks, rearranged it.